module-4/backend/week_one.md

```diff
@@ -5,18 +5,69 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's the most useful thing you learned from completing the intermission week work?
+
+learning about constructor functions and how those work
+
 2. What are some tools to help debug JavaScript code?
+
+debugger, console.log(something), broswer tools cmd + alt + i
+
 3. What are some tools you need in order to unit test your JavaScript?
+
+chai, mocha, 
+
 4. What is the syntax for invoking a function?
+
+function ()
+
 5. What is CORS?
+
+cross origin resource sharing. it is a standard for accessing web resources on different domains. 
+
 6. How do we enable CORS?
+
+For IIS6
+Open Internet Information Service (IIS) Manager.
+Right click the site you want to enable CORS for and go to Properties.
+Change to the HTTP Headers tab.
+In the Custom HTTP headers section, click Add.
+Enter Access-Control-Allow-Origin as the header name.
+Enter * as the header value.
+Click Ok twice.
+
 7. What's `this` in JavaScript?
+
+'this'is the thing that owns the javascript code. This is usually the object when used in the object itself. 
+
 8. What is Webpack and why is it useful?
+
+webpack is a module bundler. It is used in conjunction with selenium to make sure our tests are running the way we want them to. 
+
 9. When/why do you want to use event delegation?
+
+Because you might want something to happen when you clicked a place on the page, you might want something to dynamically move from one page to another. 
+
 10. What is a callback function?
+
+callback functions are ones that wait for something to happen to execute. For instance, this could be making sure that before an object is saved to the database it will make sure it has all the required fields. 
+
 11. What's `npm` and what do we use it for?
 
+npm is the js evivalent of rvm. It starts our server, runs our tests, and installs our packages. 
```

Comment:
npm is a package manage. it's the equivalent of Bundler for Ruby. it helps manage the third party packages we leverage in our code